### DISCUSSION TITLE
Docker compose error while creating mount source path

### DISCUSSION THREAD

**Discussion by tylnesh** [canonical/docker-snap#232] [General]
I happened upon an old issue with docker distributed in the snap. When using our Dockerfile with volumes defined as such:
`
     volumes:
      - type: bind
        source: /opt/something/data
        target: /opt/something/data
        bind:
          create_host_path: true
`

I get this error:
`
Error response from daemon: error while creating mount source path '/opt/something/data': mkdir /opt/something: read-only file system
`

Seems like it's the same issue as here: https://stackoverflow.com/questions/45764477/docker-compose-error-while-creating-mount-source-path

**Comment by locnnil** [comment]
As snap packages add additional layers of security, the behavior that you listed is expected.  

By default, a [strictly confined](https://snapcraft.io/docs/snap-confinement) snap (such as the Docker snap) cannot access the entire system. Instead, there are specific [data locations](https://snapcraft.io/docs/data-locations) where the snapped application is allowed access. Additionally, there are [snap interfaces](https://snapcraft.io/docs/supported-interfaces) that grant extra permissions, enabling the application inside the snap to function properly.  

The applications inside the Docker snap itself uses several of these interfaces, [like for dockerd](https://github.com/canonical/docker-snap/blob/ff3470532a4f6eccd95f578b0fdc8a63447cc9fd/snap/snapcraft.yaml#L109-L116):

https://github.com/canonical/docker-snap/blob/ff3470532a4f6eccd95f578b0fdc8a63447cc9fd/snap/snapcraft.yaml#L109-L116 As well the [Docker CLI](https://github.com/canonical/docker-snap/blob/ff3470532a4f6eccd95f578b0fdc8a63447cc9fd/snap/snapcraft.yaml#L100-L103)
https://github.com/canonical/docker-snap/blob/ff3470532a4f6eccd95f578b0fdc8a63447cc9fd/snap/snapcraft.yaml#L100-L103 and [Docker Compose](https://github.com/canonical/docker-snap/blob/ff3470532a4f6eccd95f578b0fdc8a63447cc9fd/snap/snapcraft.yaml#L123-L125):

https://github.com/canonical/docker-snap/blob/ff3470532a4f6eccd95f578b0fdc8a63447cc9fd/snap/snapcraft.yaml#L123-L125

While I understand this setup might not be ideal for development purposes, it’s designed to ensure maximum security and reliability, making it a perfect fit for production scenarios.  

Thank you for bringing this up, and I hope this explanation helps clarify things!

**Reply by charlesangus** [reply]
It's "expected" that you get an opaque and unhelpful error message which doesn't say anything about the actual issue?
